forc-pkg/src/manifest.rs

```diff
@@ -191,13 +191,13 @@ pub enum Dependency {
 #[derive(Serialize, Deserialize, Clone, Debug, Default)]
 #[serde(rename_all = "kebab-case")]
 pub struct DependencyDetails {
-    pub(crate) version: Option<String>,
+    pub version: Option<String>,
     pub path: Option<String>,
-    pub(crate) git: Option<String>,
-    pub(crate) branch: Option<String>,
-    pub(crate) tag: Option<String>,
-    pub(crate) package: Option<String>,
-    pub(crate) rev: Option<String>,
+    pub git: Option<String>,
+    pub branch: Option<String>,
+    pub tag: Option<String>,
+    pub package: Option<String>,
+    pub rev: Option<String>,
```

Comment:
Rather than making all these fields `pub` which would allow users to construct invalid `DependencyDetails` more easily, we might want to consider exposing dedicated `pub` constructor functions for each particular use case, e.g.
```rust
impl DependencyDetails {
    pub fn from_path(path: &Path) -> Self {
        // ...
    }

    pub fn from_git(repo_url: &Url, git_ref: GitReference) -> Self {
        // ...
    }

    // ... 
```

Reply:
Thanks for the feedback Mitch, quick question:
For the `from_git` method, are the `Url` and `GitReference` types here already existing in the codebase or library you are referencing, or was your intention that I make them?